### PR TITLE
LPD-43221 Create upgrade process

### PR DESCRIPTION
**Pull request by Mikellorza**
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-43221

# How am I fixing it?

Create the upgrade process to create external reference codes for the selected vocabularies, and selected display style group id. This PR is only to review it in order to contribute to the master repository when we delete the FF since we have to made the changes of this upgrade in that moment.

# How can you verify that it works?

Running added integration tests.

### CODE REVIEW

File: modules/apps/asset/asset-categories-navigation-web/src/main/java/com/liferay/asset/categories/navigation/web/internal/upgrade/v1_0_2/UpgradePortletPreferences.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+			long companyId, long ownerId, int ownerType, long plid,
+			String portletId, PortletPreferences portletPreferences)
+		throws Exception {
+
+		long displayStyleGroupId = GetterUtil.getLong(
+			portletPreferences.getValue("displayStyleGroupId", null));
+
+		if (displayStyleGroupId > 0) {
+			String scopeExternalReferenceCode = getGroupExternalReferenceCode(
+				displayStyleGroupId);
```
Comment (adolfopa) [inline modules/apps/asset/asset-categories-navigation-web/src/main/java/com/liferay/asset/categories/navigation/web/internal/upgrade/v1_0_2/UpgradePortletPreferences.java:54]: Can you rename this to `groupExternalReferenceCode`?

File: modules/apps/asset/asset-categories-navigation-web/src/main/java/com/liferay/asset/categories/navigation/web/internal/upgrade/v1_0_2/UpgradePortletPreferences.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		long[] assetVocabularyIds = GetterUtil.getLongValues(
+			StringUtil.split(assetVocabularyIdsString, ','));
+
+		if (ArrayUtil.isEmpty(assetVocabularyIds)) {
+			return;
+		}
+
+		Map<Long, List<String>> groupAssetVocabularyIdsMap =
+			_getGroupAssetVocabularyExternalReferenceCodesMap(
+				assetVocabularyIds);
```
Comment (adolfopa) [inline modules/apps/asset/asset-categories-navigation-web/src/main/java/com/liferay/asset/categories/navigation/web/internal/upgrade/v1_0_2/UpgradePortletPreferences.java:80]: Shouldn't this map be called `groupAssetVocabularyExternalReferenceCodesMap`? Usually, we name the map based on the values, not the keys.

**Review by adolfopa — CHANGES_REQUESTED** [review]
Two very minor changes. Other than that, it looks good.

File: modules/apps/asset/asset-categories-navigation-web/src/main/java/com/liferay/asset/categories/navigation/web/internal/upgrade/v1_0_2/UpgradePortletPreferences.java (hunk 1 above)
Comment (Mikellorza) [reply modules/apps/asset/asset-categories-navigation-web/src/main/java/com/liferay/asset/categories/navigation/web/internal/upgrade/v1_0_2/UpgradePortletPreferences.java]: done, thanks

**Comment by Mikellorza** [comment]
@adolfopa I changed the things you proposed. We can move forward to run the tests.

**Review by adolfopa — APPROVED** [review]

**Pull request closed without merging**